### PR TITLE
Add user profile icon support for android node

### PR DESCRIPTION
**Pull request by HenrikJannsen**
Based on https://github.com/bisq-network/bisq-mobile/pull/85

Preliminary solution. Follow up PR will come with client and iOS support.

For testing it one need to either delete the data directory in simulator or edit SplashPresenter to not skip user profile creation page.
Only here the (on node) it is applied, not at offers yet.

### CODE REVIEW

**Comment by rodvar** [comment]
rebased against main

File: androidNode/src/androidMain/kotlin/network/bisq/mobile/android/node/utils/ImageUtils.kt [hunk 1]
```diff
@@ -53,8 +54,8 @@ object ImageUtil: Logging {
 
     internal fun getImageByPath(context: Context, path: String): Bitmap? {
         return try {
-            log.d { "opening file in path $path"}
-            val inputStream = context.assets.open(path)
+            val fullPath = "cathash/$path"
```
Comment (rodvar) [inline androidNode/src/androidMain/kotlin/network/bisq/mobile/android/node/utils/ImageUtils.kt:57]: this makes the ImageUtil generic class specific to cat hashing.

Please pass the `cathash` as a baseDir parameter or as part of the path parameter

**Review by rodvar — CHANGES_REQUESTED** [review]
It works beautifully

Only 2 comments on this one, one in the code and this:

 - It would be great if we could use this PR to make the profile icon available for the rest of the app. See screenshots, the profile avatar once landed in home still hard-coded.

File: androidNode/src/androidMain/kotlin/network/bisq/mobile/android/node/utils/ImageUtils.kt (hunk 1 above)
Comment (HenrikJannsen) [reply androidNode/src/androidMain/kotlin/network/bisq/mobile/android/node/utils/ImageUtils.kt:57]: Its a specific ImageUtils for only the cathash use case. We should rename the class to make that more clear. But yes, we can move it as params.

As there will be bigger changes anyway as we need to generate the images also for client mode I would prefer to skip that for later. Probably that class will move to a KMP compatible version then.

**Comment by HenrikJannsen** [comment]
> It works beautifully
> 
> Only 2 comments on this one, one in the code and this:
> 
>     * It would be great if we could use this PR to make the profile icon available for the rest of the app. See screenshots, the profile avatar once landed in home still hard-coded.

Yes that was my plan, but wanted to get client working first. But spent yesterday the whole day to find a solution for passing the images (as its generated using javafx we cannot simply run that in headless mode, though finally i got a solution for that). But then got aware that it would be much better to do it client side as there can be many images (>1000) and it can cause quite some network load and delay. Will have a look today to port the cathash code from bisq2 to client KMP compatible version. Hope it is manageable.

File: androidNode/src/androidMain/kotlin/network/bisq/mobile/android/node/utils/ImageUtils.kt (hunk 1 above)
Comment (HenrikJannsen) [reply androidNode/src/androidMain/kotlin/network/bisq/mobile/android/node/utils/ImageUtils.kt]: Done

**Comment by HenrikJannsen** [comment]
Rebased

**Comment by rodvar** [comment]
> > It works beautifully
> > Only 2 comments on this one, one in the code and this:
> > ```
> > * It would be great if we could use this PR to make the profile icon available for the rest of the app. See screenshots, the profile avatar once landed in home still hard-coded.
> > ```
> 
> Yes that was my plan, but wanted to get client working first. But spent yesterday the whole day to find a solution for passing the images (as its generated using javafx we cannot simply run that in headless mode, though finally i got a solution for that). But then got aware that it would be much better to do it client side as there can be many images (>1000) and it can cause quite some network load and delay. Will have a look today to port the cathash code from bisq2 to client KMP compatible version. Hope it is manageable.

makes sense, let me know if I can help in any ways

**Review by rodvar — APPROVED** [review]
Great work Henrik.
I think we merged this in since is target for node, the request on using the genereated profile in the top-right account avatar can be dealt with in the next PR bringing support for the clients as well :)

**Pull request merged**